docs/001_develop/03_client-capabilities/013_toast-notifications/index.mdx

```diff
@@ -2,17 +2,364 @@
+
+By centralizing notification management, the `notification-listener` makes it easy to handle different types of notification in a consistent and modular way.
+
+The client app need to create `notification-listener` at top of the component tree. Any child under `notification-listener` component can dispatch custom **NotificationEvent** event
+
+### Example: Usage of Notification listener
+Lets look at code example to setup and use the `notification-listener` component. You will need to follow the following steps to achieve this:
+
+**1. Wrap your application with the notification-listener**
+
+As discussed earlier `notification-listener` component needs to be placed at the top of component tree to ensure it can capture events from any child component. 
+```typescript
+  <rapid-notification-listener>
+    <AppContent/>
+  </rapid-notification-listener>
+```
+
+**2. Dispatching notification using NOTIFICATION_EVENT**
+
+To dispatch notification, you can use [NotificationEvent.NOTIFICATION_EVENT](./docs/api/foundation-notifications.notificationevent.md). You can specify the type of notification component you want to display in the `config` object. For this example we are displaying snackbar.
+
+```typescript
+this.$emit( NotificationEvent.NOTIFICATION_EVENT,
+  {
+    title: 'Important message',
+    body: 'Lorem ipsum',
+    config: {
+      snackbar: {
+        confirmingActions: [{ label: 'Confirm', action: () => console.log('Lorem ipsum') }],
+        type: 'error',
+        autoClose: true
+      },
+    },
+  } as NotificationStructure,)
+```
+
+**3. Provide a UI element to trigger notification**
+
+Now you can trigger your notification on clicking on a UI element like button.
+
+
+Lets complete the above example and add a button in there to trigger the notification event.
+```typescript
+
+  import {NotificationEvent, NotificationStructure} from '@genesislcap/foundation-notifications'
+  ...
+  @customElement({
+    name: 'my-element',
+    template: html`
+    <rapid-notification-listener>
+      <div class="container">
+        <rapid-button @click=${(x) => x.triggerNotification()}>Show Notification</rapid-button>
+      </div>
+    </rapid-notification-listener>
+    `,
+  })
+  export class MyElement extends GenesisElement {
+    triggerNotification() {
+      this.$emit( NotificationEvent.NOTIFICATION_EVENT,
+        {
+          title: 'Important message',
+          body: 'Lorem ipsum',
+          config: {
+            snackbar: {
+              confirmingActions: [{ label: 'Confirm', action: () => console.log('Lorem ipsum') }],
+              type: 'error',
+              autoClose: true
+            },
+          },
+        } as NotificationStructure,)
+    }
+  }
+```
```

Comment:
These examples need react and angular examples too

Reply:
looking at this one

Reply:
added examples from both react and angular and updated text for this section